storage/innobase/xtrabackup/test/t/broken_pipe.sh

```diff
@@ -24,3 +24,27 @@ xtrabackup --backup --stream=xbstream \
 if [ ${PIPESTATUS[0]} -eq 0 ] ; then
 	die "xtrabackup completed OK when it was expected to fail"
 fi
+
+# PXB-2486: PXB hangs in case of broken pipe, --parallel
+# and --encryption or --compress
+
+timeout 20 $XB_BIN $XB_ARGS --backup --stream=xbstream \
+	--target-dir=$topdir/backup1 --parallel=10 --encrypt=AES256 \
+	--encrypt-key=2222233333232323 | dd bs=1024 count=100 of=/dev/null
+
+if [ ${PIPESTATUS[0]} -eq 124 ] ; then
+	die "xtrabackup timeout after 20 seconds when it was expected to fail"
+fi
+
+if [[ is_debug_pxb_version ]]; then
+	# let the command fail and capture the output
+	set +e
```

Comment:
whats up with this set +e and -e commands here?

Reply:
@satya-bodapati  By default, tests run with `set -e` which will fail if the command returns an error. 
At `run_cmd` and `run_cmd_expect_error` we also do change it to `set +e ... set -e`  however, here I want to capture the return code of the command to validate if it failed or timeout out (which gives 124 as return code)

Reply:
ok, special scenario